src/service.rs

```diff
@@ -389,14 +386,57 @@ impl<TYPES: NodeType> ReadState for ProxyGlobalState<TYPES> {
     }
 }
 
+pub fn broadcast_channels<TYPES: NodeType>(
+    capacity: usize,
+) -> (BroadcastSenders<TYPES>, BroadcastReceivers<TYPES>) {
+    macro_rules! pair {
+        ($s:ident, $r:ident) => {
+            let ($s, $r) = broadcast(capacity);
+            let $r = $r.deactivate();
+        };
+    }
+
+    pair!(tx_sender, tx_receiver);
+    pair!(da_sender, da_receiver);
+    pair!(qc_sender, qc_receiver);
+    pair!(decide_sender, decide_receiver);
+
+    (
+        BroadcastSenders {
+            transactions: tx_sender,
+            da_proposal: da_sender,
+            qc_proposal: qc_sender,
+            decide: decide_sender,
+        },
+        BroadcastReceivers {
+            transactions: tx_receiver,
+            da_proposal: da_receiver,
+            quorum_proposal: qc_receiver,
+            decide: decide_receiver,
+        },
+    )
+}
+
+// Receivers for HotShot events for the builder states
+pub struct BroadcastReceivers<TYPES: NodeType> {
+    /// For transactions, shared.
+    pub transactions: InactiveReceiver<Arc<ReceivedTransaction<TYPES>>>,
+    /// For the DA proposal.
+    pub da_proposal: InactiveReceiver<MessageType<TYPES>>,
+    /// For the quorum proposal.
+    pub quorum_proposal: InactiveReceiver<MessageType<TYPES>>,
+    /// For the decide.
+    pub decide: InactiveReceiver<MessageType<TYPES>>,
+}
+
 // Senders to broadcast data from HotShot to the builder states.
 pub struct BroadcastSenders<TYPES: NodeType> {
     /// For transactions, shared.
     pub transactions: BroadcastSender<Arc<ReceivedTransaction<TYPES>>>,
     /// For the DA proposal.
     pub da_proposal: BroadcastSender<MessageType<TYPES>>,
     /// For the quorum proposal.
-    pub quorum_proposal: BroadcastSender<MessageType<TYPES>>,
+    pub qc_proposal: BroadcastSender<MessageType<TYPES>>,
```

Comment:
Nit: I still think `quorum_proposal` is the better name because it's what it's called in HotShot.

Reply:
100% agree! I renamed it to `quorum_proposal` everywhere